lib/services/cloud-build-service.ts

```diff
@@ -128,70 +201,83 @@ export class CloudBuildService extends CloudService implements ICloudBuildServic
+		const buildRequestData = await this.prepareBuildRequest(projectSettings, platform, buildConfiguration, buildId, accountId, androidBuildData, iOSBuildData);
 
-		const buildResultUrl = this.getBuildResultUrl(buildResult);
-		const itmsOptions = {
-			pathToProvision: iOSBuildData && iOSBuildData.pathToProvision,
-			projectId: projectSettings.projectId,
-			projectName: projectSettings.projectName,
-			url: buildResultUrl
-		};
+		const result = await settlePromises<any>([
+			this.executeBuild(projectSettings, platform, buildRequestData, buildId, buildInformationString, iOSBuildData),
+			this.prepareProject(buildId, projectSettings, platform, buildConfiguration, iOSBuildData, projectData)
+		]);
 
-		const fullOutput = await this.getContentOfS3File(buildResponse.outputUrl);
+		const executeBuildResult = result[0];
+		const prepareProjectResult = result[1];
 
-		const result = {
-			buildId,
-			stderr: buildResult.stderr,
-			stdout: buildResult.stdout,
-			fullOutput: fullOutput,
-			outputFilePath: localBuildResult,
-			qrData: {
-				originalUrl: buildResultUrl,
-				imageData: await this.getImageData(buildResultUrl, itmsOptions)
-			}
-		};
+		if (prepareProjectResult.error) {
+			this.$logger.warn(prepareProjectResult.error.message);
+		} else {
+			const buildInfoFileDirname = path.dirname(executeBuildResult.result.outputFilePath);
+			this.$platformService.saveBuildInfoFile(platform, projectSettings.projectDir, buildInfoFileDirname);
+		}
 
-		const buildInfoFileDirname = path.dirname(result.outputFilePath);
-		this.$platformService.saveBuildInfoFile(platform, projectSettings.projectDir, buildInfoFileDirname);
-		return result;
+		if (executeBuildResult.error) {
+			this.$errors.failWithoutHelp(executeBuildResult.error.message);
+		}
+
+		return executeBuildResult.result;
 	}
 
-	public validateBuildProperties(platform: string,
-		buildConfiguration: string,
-		appId: string,
-		androidBuildData?: IAndroidBuildData,
-		iOSBuildData?: IIOSBuildData): Promise<void> {
-		return this.$nsCloudBuildPropertiesService.validateBuildProperties(platform, buildConfiguration, appId, androidBuildData, iOSBuildData);
+	private async setRuntimeVersion(projectData: IProjectData, platform: string): Promise<void> {
+		const cliVersion = this.$staticConfig.version;
+		const runtimeName = this.getRuntimeName(platform);
+		const runtimeVersion = await this.$npmInstallationManager.getLatestCompatibleVersion(runtimeName, cliVersion);
+		const packageJsonPath = path.join(projectData.projectDir, this.$constants.PACKAGE_JSON_FILE_NAME);
+		const packageJsonContent = this.$fs.readJson(packageJsonPath);
+		if (!packageJsonContent.nativescript) {
+			packageJsonContent.nativescript = {};
+		}
+
+		if (!packageJsonContent.nativescript[runtimeName]) {
+			packageJsonContent.nativescript[runtimeName] = {
+				version: runtimeVersion
+			};
+
+			this.$fs.writeJson(packageJsonPath, packageJsonContent);
+		}
```

Comment:
we need to handle the case where the `package.json` contains a tag instead of a version - for example `next` or `rc`. 
The current behavior in this case is:
1. The project preparation replaces the tag with a real version
2. Cloud build is started and the version in package.json is exact, not a tag.

With the changes in this PR, when the cloud build starts, we'll have a tag in the package.json and we may end up with different versions of runtime in the cloud and locally.